### PR TITLE
hinted handoff: Prevent segmentation fault when initializing endpoint managers 

### PR DESCRIPTION
**Pull request by dawmd**
We don't attempt to create an endpoint manager for a hint directory if there is no mapping host ID–IP corresponding to the directory's name, an IP address. That prevents a segmentation fault.

Fixes scylladb/scylladb#18649

### CODE REVIEW

File: db/hints/manager.cc
Comment (piodul) [inline db/hints/manager.cc:876]: Spelling fixes are no longer needed, they were fixed by https://github.com/scylladb/scylladb/issues/18567 which is already on master

**Comment by dawmd** [comment]
Version 2:
* dropped the commits fixing spelling

**Review by piodul — APPROVED** [review]
LGTM, will wait for CI before merging

**Comment by piodul** [comment]
Queued

**Pull request merged**
